staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr_test.go

```diff
@@ -110,6 +110,87 @@ func TestIntOrStringMarshalJSONUnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestGetIntFromString(t *testing.T) {
+	tests := []struct {
+		input      IntOrString
+		expectErr  bool
+		expectVal  int
+		expectPerc bool
+	}{
+		{
+			input:      FromInt(200),
+			expectErr:  false,
+			expectVal:  200,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("200"),
+			expectErr:  false,
+			expectVal:  200,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("30%0"),
+			expectErr:  true,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("40%"),
+			expectErr:  false,
+			expectVal:  40,
+			expectPerc: true,
+		},
+		{
+			input:      FromString("%"),
+			expectErr:  true,
+			expectPerc: true,
+		},
+		{
+			input:      FromString("a%"),
+			expectErr:  true,
+			expectPerc: true,
+		},
+		{
+			input:      FromString("a"),
+			expectErr:  true,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("40#"),
+			expectErr:  true,
+			expectPerc: false,
+		},
+	}
+	for i, test := range tests {
+		t.Logf("test case %d", i)
+		value, isPercent, err := getIntOrPercentValue(&test.input)
+		if test.expectVal != value {
+			t.Errorf("expected value does not match")
+			continue
+		}
+		if test.expectPerc && isPercent == false {
+			t.Errorf("expected percentage, but got not percentage")
+			continue
+		}
+		if !test.expectPerc && isPercent == true {
+			t.Errorf("expected not percentage and got percentage")
+			continue
+		}
+		if test.expectErr && err == nil {
+			t.Errorf("expected error, but got none")
+			continue
+		}
+		if !test.expectErr && err != nil {
+			t.Errorf("unexpected err: %v", err)
+			continue
+		}
+		if test.expectVal != value {
+			t.Errorf("expected %v, but got %v", test.expectVal, value)
+		}
+	}
```

Comment:
This could be condensed a lot, I'd also recommend using `t.Run` so execute a subtest here, then instead of Error and Continue you can Fatal the subtest
```suggestion
	for _, test := range tests {
	    t.Run("", func(t *testing.T) {
	        value, isPercent, err := getIntOrPercentValue(&test.input)
		if test.expectVal != value {
			t.Fatalf("expected value does not match, expected: %d, got: %d", test.expectVal, value)
		}
		if test.expectPerc != isPercent {
			t.Fatalf("expected percent does not match, expected: %d, got: %d", test.expectPerc, isPercent)
		}
		if test.expectErr != (err != nil) {
			t.Fatalf("expected error does not match, expected error: %v, got: %v", test.expectErr, err)
		}
	    })	
	}
```

Reply:
Looks good to me, commited the suggestion